Shared/Samples/Style point with picture marker symbols/StylePointWithPictureMarkerSymbolsView.swift

```diff
@@ -0,0 +1,90 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct StylePointWithPictureMarkerSymbolsView: View {
+    /// The graphics overlay to hold the point graphics.
+    @State private var graphicsOverlay: GraphicsOverlay = {
+        let graphicsOverlay = GraphicsOverlay()
+        return graphicsOverlay
+    }()
+    
+    /// A map with topographic basemap and centered on Harman's Cross in England.
+    @State private var map: Map = {
+        let map = Map(basemapStyle: .arcGISTopographic)
+        map.initialViewpoint = Viewpoint(
+            center: Point(x: -225166.5, y: 6551249, spatialReference: .webMercator),
+            scale: 1e5
+        )
+        
+        return map
+    }()
+    
+   /// Adds a picture maker from an image in the project assets.
+    private func addPictureMarkerSymbolFromImage() {
+        let imageName = "PinBlueStar"
+        
+        // Create pin symbol using the image.
+        let pinSymbol = PictureMarkerSymbol(image: UIImage(named: imageName)!)
+        
+        // Change offsets, so the symbol aligns properly to the point.
+        pinSymbol.offsetY = pinSymbol.image!.size.height / 2
+        
+        // Create the location for pin.
+        let pinPoint = Point(x: -226773, y: 6550477, spatialReference: .webMercator)
+        
+        // Create the graphic for pin.
+        let pinGraphic = Graphic(geometry: pinPoint, symbol: pinSymbol)
+        
+        // Add the graphic to the overlay.
+        self.graphicsOverlay.addGraphic(pinGraphic)
+    }
+    
+    /// Adds a picture marker using a remote image.
+    private func addPictureMarkerSymbolFromURL() {
+        let imageURL = URL(string: "https://static.arcgis.com/images/Symbols/OutdoorRecreation/Camping.png")!
+            
+        // Create pin symbol using the URL.
+        let campsiteSymbol = PictureMarkerSymbol(url: imageURL)
+            
+        // Optionally set the size.
+        // (If not set, the size in pixels of the image will be used.)
+        campsiteSymbol.width = 24
+        campsiteSymbol.height = 24
+            
+        // Create the location for campsite.
+        let campsitePoint = Point(x: -223560, y: 6552021, spatialReference: .webMercator)
+            
+        // Create the graphic for campsite.
+        let campsiteGraphic = Graphic(geometry: campsitePoint, symbol: campsiteSymbol)
+            
+        // Add the graphic to the overlay.
+        self.graphicsOverlay.addGraphic(campsiteGraphic)
+    }
+    
+    init() {
+        // Add picture marker symbol using a remote image.
+        self.addPictureMarkerSymbolFromURL()
+        
+        // Add picture marker symbol using image in assets.
+        self.addPictureMarkerSymbolFromImage()
+    }
```

Comment:
It should not be a problem, but until Apple provides a doc that explains the lifecycle of SwiftUI (like [UIViewController](https://developer.apple.com/documentation/uikit/uiviewcontroller#1652793)), I would refrain setting anything up that depends on a SwiftUI component, in the initializer of a SwiftUI View. Here because the graphics overlay is a state variable, it might have unknown side effects. Putting it in the task block is a bit safer.

```swift
        MapView(map: map, graphicsOverlays: [graphicsOverlay])
            .task {
                // Add picture marker symbol using a remote image.
                addPictureMarkerSymbolFromURL()
                
                // Add picture marker symbol using image in assets.
                addPictureMarkerSymbolFromImage()
            }
```

Reply:
As a value type, a SwiftUI view instance doesn't have a lifecycle per se. A view is instantiated whenever the parent's body is computed, which can happen any number of times. As part of the initializer, this code would add the two graphics to the graphics overlay whenever the parent view is re-computed, which isn't great. While using a `task` modifier would prevent that from happening, but since these aren't asynchronous methods, that would be an abuse of that modifier as it is designed for performing asynchronous operations. Instead, the actions performed by these two methods should just be part of creating the graphics overlay. Something like this:

```swift
@State private var graphicsOverlay: GraphicsOverlay = {
    let graphics = [
        makePictureMarkerSymbolGraphicFromImage(),
        makePictureMarkerSymbolGraphicFromURL()
    ]
    return GraphicsOverlay(graphics: graphics)
}()
```

private static func makePictureMarkerSymbolGraphicFromImage() -> Graphic { ... }

private static func makePictureMarkerSymbolGraphicFromURL() -> Graphic { ... }

Reply:
Agreed. Using type property or type factory method make sense.